Crash on if nested in for `in`.

### Bug description

When parsing the following (excerpt of a) file:

<!--
 If sharing the code is not an option, please state so,
 but providing only the stacktrace would still be helpful.
 -->

```python
for index, _ in enumerate(lijstv) if len(lijst1) >= len(lijstv) else enumerate(lijst1):
    a = waarde2verschil(lijst1[index] - lijstv[index])

if len(lijst1) < len(lijstv):
    for index2 in range(vers):
        a = waarde2verschil(lijstv[index + index2 + 1])


```

pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 727, in _check_file
    check_astroid_module(ast_node)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 926, in check_astroid_module
    retval = self._check_astroid_module(
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 976, in _check_astroid_module
    walker.walk(node)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  [Previous line repeated 5 more times]
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1349, in visit_name
    self._undefined_and_used_before_checker(node, stmt)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1391, in _undefined_and_used_before_checker
    action, nodes_to_consume = self._check_consumer(
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1496, in _check_consumer
    self._loopvar_name(node)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 2242, in _loopvar_name
    and assign.iter.args
AttributeError: 'IfExp' object has no attribute 'args'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 684, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 729, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Configuration

```ini
No configuration.
```


### Command used

```shell
pylint 9860988.py
```


### Pylint output

```shell
************* Module 9860988
9860988.py:7:0: C0305: Trailing newlines (trailing-newlines)
9860988.py:1:0: C0114: Missing module docstring (missing-module-docstring)
9860988.py:1:0: C0103: Module name "9860988" doesn't conform to snake_case naming style (invalid-name)
9860988.py:1:41: E0602: Undefined variable 'lijst1' (undefined-variable)
9860988.py:1:56: E0602: Undefined variable 'lijstv' (undefined-variable)
9860988.py:1:26: E0602: Undefined variable 'lijstv' (undefined-variable)
9860988.py:1:79: E0602: Undefined variable 'lijst1' (undefined-variable)
9860988.py:2:8: E0602: Undefined variable 'waarde2verschil' (undefined-variable)
9860988.py:2:24: E0602: Undefined variable 'lijst1' (undefined-variable)
9860988.py:2:40: E0602: Undefined variable 'lijstv' (undefined-variable)
9860988.py:4:7: E0602: Undefined variable 'lijst1' (undefined-variable)
9860988.py:4:21: E0602: Undefined variable 'lijstv' (undefined-variable)
9860988.py:5:24: E0602: Undefined variable 'vers' (undefined-variable)
9860988.py:6:12: E0602: Undefined variable 'waarde2verschil' (undefined-variable)
9860988.py:6:28: E0602: Undefined variable 'lijstv' (undefined-variable)
Exception on node <Name.index l.6 at 0x7fc6a50d2440> in file '/home/charlotte/repos/dodona/themis/9860988.py'
Traceback (most recent call last):
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1349, in visit_name
    self._undefined_and_used_before_checker(node, stmt)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1391, in _undefined_and_used_before_checker
    action, nodes_to_consume = self._check_consumer(
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1496, in _check_consumer
    self._loopvar_name(node)
  File "/nix/store/sym07z6bfly57vim5fxmf41xw8knlp25-python3-3.10.5-env/lib/python3.10/site-packages/pylint/checkers/variables.py", line 2242, in _loopvar_name
    and assign.iter.args
AttributeError: 'IfExp' object has no attribute 'args'
9860988.py:1:0: F0002: 9860988.py: Fatal error while checking '9860988.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/charlotte/.cache/pylint/pylint-crash-2022-07-05-15.txt'. (astroid-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### Expected behavior

It should not crash.

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.10.5 (main, Jun  6 2022, 12:05:50) [GCC 11.3.0]
```


### OS / Environment

NixOS unstable.

### Additional dependencies

_No response_

--- a/doc/whatsnew/2/2.14/full.rst
+++ b/doc/whatsnew/2/2.14/full.rst
@@ -6,6 +6,11 @@ What's New in Pylint 2.14.5?
 Release date: TBA
 
 
+* Fixed a crash in the ``undefined-loop-variable`` check when ``enumerate()`` is used
+  in a ternary expression.
+
+  Closes #7131
+
 * Fixed handling of ``--`` as separator between positional arguments and flags.
 
   Closes #7003

--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -2237,9 +2237,12 @@ class VariablesChecker(BaseChecker):
             if (
                 isinstance(inferred, astroid.Instance)
                 and inferred.qname() == "builtins.enumerate"
-                and assign.iter.args
             ):
-                inferred = next(assign.iter.args[0].infer())
+                likely_call = assign.iter
+                if isinstance(assign.iter, nodes.IfExp):
+                    likely_call = assign.iter.body
+                if isinstance(likely_call, nodes.Call):
+                    inferred = next(likely_call.args[0].infer())
         except astroid.InferenceError:
             self.add_message("undefined-loop-variable", args=node.name, node=node)
         else:

--- a/tests/functional/u/undefined/undefined_loop_variable.py
+++ b/tests/functional/u/undefined/undefined_loop_variable.py
@@ -148,6 +148,13 @@ def use_enumerate():
     print(i, num)
 
 
+def use_enumerate_in_ternary_expression():
+    """https://github.com/PyCQA/pylint/issues/7131"""
+    for i, num in enumerate(range(3)) if __revision__ else enumerate(range(4)):
+        pass
+    print(i, num)
+
+
 def find_even_number(container):
     """https://github.com/PyCQA/pylint/pull/6923#discussion_r895134495"""
     for something in container:

--- a/tests/functional/u/undefined/undefined_loop_variable.txt
+++ b/tests/functional/u/undefined/undefined_loop_variable.txt
@@ -1,4 +1,4 @@
 undefined-loop-variable:6:11:6:14:do_stuff:Using possibly undefined loop variable 'var':UNDEFINED
 undefined-loop-variable:25:7:25:11::Using possibly undefined loop variable 'var1':UNDEFINED
 undefined-loop-variable:75:11:75:14:do_stuff_with_redefined_range:Using possibly undefined loop variable 'var':UNDEFINED
-undefined-loop-variable:156:11:156:20:find_even_number:Using possibly undefined loop variable 'something':UNDEFINED
+undefined-loop-variable:163:11:163:20:find_even_number:Using possibly undefined loop variable 'something':UNDEFINED

